### PR TITLE
[6.1] OD bounding box co-ordinates cannot contain None values

### PR DESCRIPTION
**Pull request by jakesabathia2**
https://github.com/apple/turicreate/issues/2833

### CODE REVIEW

File: src/toolkits/object_detection/od_data_iterator.cpp [hunk 1]
```diff
@@ -121,6 +121,16 @@ std::vector<image_annotation> parse_annotations(
...
 
           const flex_string& coord_key = coord_kv.first.get<flex_string>();
+
+          // Check the type of coordinate
+          if (coord_kv.second.get_type() != flex_type_enum::FLOAT and
+              coord_kv.second.get_type() != flex_type_enum::INTEGER) {
+            log_and_throw(
+                "Invalid object annotations discovered. Annotation coordinate "
+                "'" +
+                coord_key + "' must have type of Int or Float.");
```
Comment (srikris) [inline src/toolkits/object_detection/od_data_iterator.cpp]: "Bounding box co-ordinates must be int or float values" is more accurate.

Comment (jakesabathia2) [reply src/toolkits/object_detection/od_data_iterator.cpp]: good!

File: src/toolkits/object_detection/od_data_iterator.cpp [hunk 2]
```diff
@@ -125,6 +125,14 @@ std::vector<image_annotation> parse_annotations(
         for (const auto& coord_kv : coordinates) {
 
           const flex_string& coord_key = coord_kv.first.get<flex_string>();
+
+          // Check the type of coordinate
```
Comment (Jarvi-Izana) [inline src/toolkits/object_detection/od_data_iterator.cpp]: good job. One more thing, please update the docstring if the type constraint of coordinates is not mentioned.

Comment (Jarvi-Izana) [reply src/toolkits/object_detection/od_data_iterator.cpp]: also, don't forget to check the width and height. ;-)

Comment (jakesabathia2) [reply src/toolkits/object_detection/od_data_iterator.cpp]: It does include width and height ;)

Comment (Jarvi-Izana) [reply src/toolkits/object_detection/od_data_iterator.cpp]: Ah, it's a map. Good!

**Pull request merged**